oauth2/handler.go

```diff
@@ -308,9 +308,19 @@ func (h *Handler) UserinfoHandler(w http.ResponseWriter, r *http.Request) {
 	delete(interim, "sid")
 	delete(interim, "jti")
 
-	if aud := interim["aud"].([]string); !ok || len(aud) == 0 {
+	if aud, ok := interim["aud"].([]string); !ok || len(aud) == 0 {
 		interim["aud"] = []string{c.GetID()}
 	}
+	found := false
+	for _, a := range interim["aud"].([]string) {
+		if a == c.GetID() {
+			found = true
+			break
+		}
+	}
+	if !found {
```

Comment:
It would probably make sense to type assert once and set the value on interim later. You might use ory/x/stringslice for convenience!

Reply:
Updated. I didn't find an opportunity to use ory/x/stringslice though.